xarray/tests/test_variable.py

```diff
@@ -294,6 +294,12 @@ def test_object_conversion(self):
         actual = self.cls("x", data)
         assert actual.dtype == data.dtype
 
+    def test_datetime64_valid_range(self):
+        data = np.datetime64("1250-01-01", "us")
+        pderror = pd._libs.tslibs.np_datetime.OutOfBoundsDatetime
```

Comment:
I think https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.errors.OutOfBoundsDatetime.html is the public API